### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by cjwatson**
We can't run PyPy3 tests there yet, because PyPy 7.2.0 and later break
zope.password there (https://foss.heptapod.net/pypy/pypy/-/issues/3378),
while older versions aren't available on GHA.

### CODE REVIEW

**Comment by cjwatson** [comment]
You can see a test run of more or less this commit [here](https://github.com/cjwatson/zopetoolkit/actions/runs/462346102).

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
```
Comment (mgedmin) [inline .github/workflows/tests.yml:2]: Is it really?

Maybe rephrase as `# Based on` or `# Inspired by`?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        - ["2.7",      "setuptools==44.0.0"]
+        - ["3.6",      "setuptools"]
+        - ["3.7",      "setuptools"]
+        - ["3.8",      "setuptools"]
+        - ["3.9",      "setuptools"]
+        - ["pypy-2.7", "setuptools==44.0.0"]
+        # Temporarily disabled because
+        # https://foss.heptapod.net/pypy/pypy/-/issues/3378 breaks
+        # zope.password
+        # - ["pypy-3.6", "setuptools"]
```
Comment (mgedmin) [inline .github/workflows/tests.yml:25]: Can we run the tests, but allow pypy3 to fail?

AFAIU https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error is how you do it:

```yaml
    runs-on: ubuntu-latest
    name: ${{ matrix.config[0] }}
    continue-on-error: ${{ matrix.config[0] == 'pypy-3.6' }}
    steps:
        ...
```

**Review by mgedmin — APPROVED** [review]
LGTM with suggestions

File: .github/workflows/tests.yml (hunk 1 above)
Comment (cjwatson) [reply .github/workflows/tests.yml:2]: Oops.  Adjusted, thanks.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (cjwatson) [reply .github/workflows/tests.yml]: I went for setting `continue-on-error` in the relevant `steps` entry instead, because that way PRs that pass everywhere other than pypy3 will actually show up with a tick rather than maintainers having to dig into the details to check for regressions.  (A more surgical alternative would be to exclude just `zope.password` from just the pypy3 tests, but I couldn't work out how to do that with `z3c.recipe.compattest`.  Maybe somebody else can.)
